tests/MyGame/Example/Monster.java

```diff
@@ -14,6 +14,9 @@
 public final class Monster extends Table {
   public static Monster getRootAsMonster(ByteBuffer _bb) { return getRootAsMonster(_bb, new Monster()); }
   public static Monster getRootAsMonster(ByteBuffer _bb, Monster obj) { _bb.order(ByteOrder.LITTLE_ENDIAN); return (obj.__assign(_bb.getInt(_bb.position()) + _bb.position(), _bb)); }
+  public static Monster getSizePrefixedRootAsMonster(ByteBuffer _psbb) { return getSizePrefixedRootAsMonster(_psbb, new Monster()); }
+  public static Monster getSizePrefixedRootAsMonster(ByteBuffer _psbb, Monster obj) { ByteBuffer _bb = _psbb.slice(); _bb.position(4); return getRootAsMonster(_bb, obj); }
```

Comment:
Not sure why we're creating a new `ByteBuffer` here, ideally this refers to the existing one?

Reply:
There is a new buffer created, however `slice()` uses the data of the other buffer, there is no copying involved, only position, limit etc. are reset. The same happens with the `Slice()` version I have added to the C# `ByteBuffer`. My reasoning behind this was that I wanted to avoid advancing the passed-in `_psbb` to avoid side-effects for the user. One possibility would be to increment the position by 4, do the regular `getRootAsMonster`, and then decrement it again. However I was worried about concurrent access to the buffer. Any thoughts on this? Would the +4, `getRootAsMonster`, -4 version be okay with you?

Reply:
I'm aware we're not copying the buffer, but I generally like to avoid object allocations where possible.
I guess the + then - trick would be ok, but really we shouldn't need to touch the `position` at all if you see how the existing `getRootAsMonster` works, you can just generate a similar method that does `_bb.position() + 4` on the fly.